File: Handy/Binary Search O(log n)/recursive.js
//Recursive
function binarySearchRecursive(arr, target, left, right) {
    if (left > right) {
        return -1;  // Target not found
    }

    let mid = Math.floor((left + right) / 2);

    if (arr[mid] === target) {
        return mid;
    } else if (arr[mid] < target) {
        return binarySearchRecursive(arr, target, mid + 1, right);
    } else {
        return binarySearchRecursive(arr, target, left, mid - 1);
    }
}

// Example usage
let arr = [1, 2, 3, 4, 5, 6, 7];
let target = 5;
console.log(binarySearchRecursive(arr, target, 0, arr.length - 1));  // Output: 4

File: Handy/Binary Search O(log n)/recursive.py
#Recursive
def binary_search_recursive(arr, target, left, right):
    if left > right:
        return -1  # Target not found
    
    mid = (left + right) // 2
    
    if arr[mid] == target:
        return mid
    elif arr[mid] < target:
        return binary_search_recursive(arr, target, mid + 1, right)
    else:
        return binary_search_recursive(arr, target, left, mid - 1)

# Example usage
arr = [1, 2, 3, 4, 5, 6, 7]
target = 5
print(binary_search_recursive(arr, target, 0, len(arr) - 1))  # Output: 4
